client/state/data-layer/wpcom/account-recovery/request-reset/test/index.js

```diff
@@ -14,32 +15,47 @@ import useNock from 'test/helpers/use-nock';
 import {
 	ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
 	ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
+	ACCOUNT_RECOVERY_RESET_SET_METHOD,
 } from 'state/action-types';
 
 describe( 'handleRequestReset()', () => {
-	const dispatch = sinon.spy();
-
 	const apiBaseUrl = 'https://public-api.wordpress.com:443';
 	const endpoint = '/wpcom/v2/account-recovery/request-reset';
 
-	const request = {
-		user: 'foo',
-		method: 'primary-email',
-	};
+	const userData = { user: 'foo' };
+	const method = 'primary-email';
 
 	describe( 'success', () => {
 		useNock( nock => (
 			nock( apiBaseUrl )
+				.persist()
 				.post( endpoint )
 				.reply( 200, { success: true } )
 		) );
 
 		it( 'should dispatch SUCCESS action on success', () => {
-			return handleRequestReset( { dispatch }, { request } ).then( () =>
-				assert.isTrue( dispatch.calledWith( {
-					type: ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
-				} ) )
-			);
+			const dispatch = sinon.spy( ( action ) => {
+				if ( action.type === ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS ) {
+					assert.isTrue( dispatch.calledWith( {
+						type: ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+					} ) );
+				}
+			} );
+
+			handleRequestReset( { dispatch }, { userData, method }, noop );
+		} );
+
+		it( 'should dispatch SET_METHOD action on success', () => {
+			const dispatch = sinon.spy( ( action ) => {
+				if ( action.type === ACCOUNT_RECOVERY_RESET_SET_METHOD ) {
```

Comment:
This test and the one [above](https://github.com/Automattic/wp-calypso/pull/13360/files#diff-d0a8b1a12a8b2c0b95fd5862f90500c8R36) will pass when it shouldn't if the `ACCOUNT_RECOVERY_RESET_SET_METHOD` is never dispatched. The fact that you are checking with `action.type === ACCOUNT_RECOVERY_RESET_SET_METHOD` is negating the entire test so that it will *mostly* pass.